obspy/io/sac/tests/test_core.py

```diff
@@ -316,7 +316,7 @@ def test_undefinedB(self):
             self.assertEqual(tr2.stats.starttime.timestamp, 269596810.0)
```

Comment:
It is not clear to me what `test_undefinedB` is testing, and I think it may need to change.  This test seems to make sure that unsetting tr.stats.sac.b doesn't change the tr.stats.starttime with the trace is written and re-read in. It does this, however, by changing the SAC reference time ("nz" header fields).  You can see this in `self.assertEqual(sac_ref_time.timestamp, 269596800.0)` and `self.assertEqual(sac_ref_time2.timestamp, 269596810.0)`.  Changing the SAC reference time is not OK, because it invalidates all other relative time headers in the file (i.e. a, o, and t0-t9, which are commonly used to store a number of phase picks and windows).  Also, if the file is iztype "io", this is equivalent to changing an event origin time.  This test preserves tr.stats.starttime when "b" is unset only by invalidating the SAC file.  I think there is a better way.

When writing a SAC file from a Trace that was also read in from a SAC file, the old tr.stats.sac.b doesn't matter.  One only has to check that the old tr.stats.sac header has a valid reference time, compare it to tr.stats.starttime, and then write a new "b" (and "e") that respects this difference.  No `iztype` checking or changes to the SAC reference time is required, and any relative time headers therein are preserved.  

I would like to change this test such that it (1) makes sure tr.stats.starttime does not change even if "b" is unset, (2) makes sure the SAC reference time ("sac_ref_time") also does not change event if "b" is unset, and (3) allows "b" to become set when writing from a Trace object.